test/statement.test.ts

```diff
@@ -0,0 +1,532 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import assert, {
+  deepStrictEqual,
+  match,
+  rejects,
+  strictEqual,
+  throws,
+} from "assert";
+import { describe, test } from "mocha";
+import { getAccounts } from "@tableland/local";
+import { getDelay } from "../src/helpers/utils.js";
+import { overrideDefaults, getDefaultProvider } from "../src/helpers/index.js";
+import { Database, Statement } from "../src/index.js";
+
+overrideDefaults("localhost", {
+  contractAddress: "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
+});
+
+describe("statement", function () {
+  // Note that we're using the second account here
+  const [, wallet] = getAccounts();
+  const provider = getDefaultProvider("http://127.0.0.1:8545");
+  const signer = wallet.connect(provider);
+  const db = new Database({ signer });
+
+  test("when initialized via constructor", async function () {
+    const sqlString = "CREATE TABLE test (counter integer);";
+    const stmt = new Statement(db.config, sqlString);
+    const { sql, parameters } = stmt.toObject();
+    strictEqual(stmt.toString(), sqlString);
+    strictEqual(parameters, undefined);
+    strictEqual(sql, sqlString);
+  });
+
+  test("when created via db.prepare()", async function () {
+    const sql = "CREATE TABLE test (counter integer);";
+    const stmt = db.prepare(sql);
+    strictEqual(stmt.toString(), sql);
+    deepStrictEqual(stmt, new Statement(db.config, sql));
+  });
+
+  describe(".run()", function () {
+    let tableName: string;
+    this.beforeAll(async function () {
+      this.timeout("10s");
```

Comment:
Most of these beforeAll hooks have a 10s timeout because they involve waiting for a create and insert to complete.